wpilibNewCommands/src/main/native/include/frc2/command/MecanumFollowerCommand.h

```diff
@@ -0,0 +1,183 @@
+ * secondary constructor that omits the PID and feedforward functionality,
+ * returning only the raw wheel speeds from the PID controllers.
+ *
+ * <p>The robot angle controller does not follow the angle given by
+ * the trajectory but rather goes to the angle given in the final state of the
+ * trajectory.
+ */
+class MecanumFollowerCommand
+    : public CommandHelper<CommandBase, MecanumFollowerCommand> {
+ public:
+  /**
+   * Constructs a new MecanumFollowerCommand that when executed will follow the
+   * provided trajectory. PID control and feedforward are handled internally,
+   * outputs are scaled from -12 to 12 as a voltage output to the motor.
+   *
+   * <p>Note: The controllers will *not* set the outputVolts to zero upon
+   * completion of the path this is left to the user, since it is not
+   * appropriate for paths with nonstationary endstates.
+   *
+   * <p>Note2: The rotation controller will calculate the rotation based on the
+   * final pose in the trajectory, not the poses at each time step.
+   *
+   * @param trajectory                        The trajectory to follow.
+   * @param pose                              A function that supplies the robot
+   * pose - use one of the odometry classes to provide this.
+   * @param feedforward                       The feedforward to use for the
+   * drivetrain.
+   * @param kinematics                        The kinematics for the robot
+   * drivetrain.
+   * @param xController                       The Trajectory Tracker PID
+   * controller for the robot's x position.
+   * @param yController                       The Trajectory Tracker PID
+   * controller for the robot's y position.
+   * @param thetaController                   The Trajectory Tracker PID
+   * controller for angle for the robot.
+   * @param maxWheelVelocityMetersPerSecond   The maximum velocity of a
+   * drivetrain wheel.
+   * @param currentWheelSpeeds                A MecanumDriveWheelSpeeds object
+   * containing
+   * @param frontLeftController               The front left wheel velocity PID.
+   * @param rearLeftController                The rear left wheel velocity PID.
+   * @param frontRightController              The front right wheel velocity
+   * PID.
+   * @param rearRightController               The rear right wheel velocity PID.
+   *                                          the current wheel speeds.
+   * @param output                            The output of the velocity PIDs
+   * and feedfrowards in volts.
+   * @param requirements                      The subsystems to require.
+   */
+  MecanumFollowerCommand(
+      frc::Trajectory trajectory, std::function<frc::Pose2d()> pose,
+      frc::SimpleMotorFeedforward<units::meters> feedforward,
+      frc::MecanumDriveKinematics kinematics, frc2::PIDController xController,
+      frc2::PIDController yController,
+      frc::ProfiledPIDController thetaController,
+      units::meters_per_second_t MaxWheelVelocityMetersPerSecond,
```

Comment:
The variable name shouldn't start with a capital letter. Furthermore, it shouldn't say the unit. That's already obvious from the type. We only did this in Java because it doesn't have a good units system.